### PR TITLE
update build for dax_imgserv

### PR DESCRIPTION
**Pull request by kennywlo**
Please review changes: 1) remove obsolete dax/webserv build 2) enhance docker build for imgserv.

### CODE REVIEW

File: pipelines/dax/docker/build_imgserv.groovy [hunk 1]
```diff
@@ -13,28 +13,55 @@ node('jenkins-master') {
...
+    dir('imgserv/lsst-dm-ci') {
+        stage('build') {
+            retry(retries) {
+                build(repo)
+            }
+        } // build
+
+        stage('test') {
+            retry(retries) {
+                docker.image(repo:tag).inside {
```
Comment (jhoblitt) [inline pipelines/dax/docker/build_imgserv.groovy]: I assume there is a missing string to be interpolated into.

Comment (jhoblitt) [inline pipelines/dax/docker/build_imgserv.groovy]: Is there a missing `dir` block here?

File: pipelines/dax/docker/build_imgserv.groovy [hunk 2]
```diff
@@ -13,28 +13,55 @@ node('jenkins-master') {
...
+                    util.bash "./run_tests.sh"
+                }
+            }
+        } // test
+
+        stage('publish') {
+            util.bash "./pub_image.sh"
```
Comment (jhoblitt) [inline pipelines/dax/docker/build_imgserv.groovy]: Is there a missing dir block here?

File: pipelines/dax/docker/build_imgserv.groovy [hunk 3]
```diff
@@ -13,28 +13,55 @@ node('jenkins-master') {
...
+    timeout(time: 2, unit: 'HOURS') {
       dir('imgserv') {
         git([
           url: util.githubSlugToUrl('lsst/dax_imgserv'),
           branch: 'master'
         ])
       }
-
-      build('prod_image.sh')
+      try {
```
Comment (jhoblitt) [inline pipelines/dax/docker/build_imgserv.groovy]: This try statement is unnecessary.

Comment (kennywlo) [inline pipelines/dax/docker/build_imgserv.groovy]: If try statement is to be removed, which I don't mind, how should I go about setting currentBuild.result?

File: pipelines/dax/docker/build_imgserv.groovy (hunk 2 above)
Comment (kennywlo) [reply pipelines/dax/docker/build_imgserv.groovy]: I was trying to encapsulate all 3 stages in a single dir block - not the right syntax?

File: pipelines/dax/docker/build_imgserv.groovy (hunk 1 above)
Comment (kennywlo) [reply pipelines/dax/docker/build_imgserv.groovy]: See comment below on dir.

Comment (kennywlo) [reply pipelines/dax/docker/build_imgserv.groovy]: is '$repo:$tag' the correct syntax for the missing string?

File: pipelines/dax/docker/build_imgserv.groovy (hunk 2 above)
Comment (jhoblitt) [reply pipelines/dax/docker/build_imgserv.groovy]: Oh, I thought the dir block was closed with the first stage.  The way it is should be fine.

File: pipelines/dax/docker/build_imgserv.groovy (hunk 3 above)
Comment (jhoblitt) [reply pipelines/dax/docker/build_imgserv.groovy]: It doesn't need to be set at all unless the failure exception is being caught and not rethrown.

File: pipelines/dax/docker/build_imgserv.groovy (hunk 1 above)
Comment (kennywlo) [reply pipelines/dax/docker/build_imgserv.groovy]: See latest. I'm now using "${repo}:${tag}" for string interpolation.

File: pipelines/dax/docker/build_imgserv.groovy (hunk 3 above)
Comment (kennywlo) [reply pipelines/dax/docker/build_imgserv.groovy]: Try block removed.

Comment (kennywlo) [reply pipelines/dax/docker/build_imgserv.groovy]: Removed.

**Comment by cbanek** [comment]
Sorry, I'm a bit out of my depth here in terms of the Jenkins parts.  We're removing the old one, which I understand, is the new one just being scheduled here with this change?

**Comment by kennywlo** [comment]
@cbanek See DM-20192.  The new one was set up to run prior, but now is much enhanced with this change.

**Comment by cbanek** [comment]
Ah okay, yes knowing the ticket is helpful.  I think it is standard practice to tie the ticket name to the branch name or something in the commit message to link everything together.

**Comment by kennywlo** [comment]
Now I see what you mean (even for a forked repo) looking at the commits history of this repo.   I will try that route next time, tying in the ticket with the PR.

File: pipelines/dax/docker/build_imgserv.groovy [hunk 4]
```diff
@@ -13,28 +13,50 @@ node('jenkins-master') {
...
       dir('imgserv') {
         git([
           url: util.githubSlugToUrl('lsst/dax_imgserv'),
           branch: 'master'
         ])
       }
-
-      build('prod_image.sh')
+      run()
+      currentBuild.result = 'SUCCESS'
```
Comment (jhoblitt) [inline pipelines/dax/docker/build_imgserv.groovy]: Do not set `currentBuild`, it isn't necessary.

File: pipelines/dax/docker/build_imgserv.groovy (hunk 1 above)
Comment (jhoblitt) [reply pipelines/dax/docker/build_imgserv.groovy]: Looks OK now.

File: pipelines/dax/docker/build_imgserv.groovy (hunk 3 above)
Comment (jhoblitt) [reply pipelines/dax/docker/build_imgserv.groovy]: OK

**Pull request merged**